### ISSUE TITLE
i7-5775C is locked to 3.6ghz even overclocking

### ISSUE THREAD

**Issue by aa15032261** [Piker-Alpha/ssdtPRGen.sh#277]
Hello, I just noticed that my 5775C is locked to 3.6ghz even I am overclocking it to 4.0ghz.

I use this command to generate ssdt for my cpu, it allows my igpu run at full speed but not cpu.
~/ssdtPRGen.sh -f 4000

I check the frequency in intel power gadget but it never run above 3.6ghz. Btw, is it possible to overclock igpu in osx?

![screen shot 2016-10-05 at 11 41 41 am](https://cloud.githubusercontent.com/assets/22629802/19100355/ca370a36-8af0-11e6-8d0a-41cc1ab7a8b8.png)

**Comment by Piker-Alpha** [comment]
Did you actually change the turbo settings in the BIOS? If not, then you should do this since it is not enough to run the script with the OC frequency.

**Comment by aa15032261** [comment]
Yes, I set all cores to 4.0ghz then install the ssdt to acpi/patched/ but my cpu cannot run above 3.6ghz. What else can I try?

**Comment by Piker-Alpha** [comment]
Please note that (some) Broadway plists (think /S_/L_/E_/IOPlatformPluginFamily.kext/C_/P_/X86PlatformPlugin.kext/C_/R_/_.plist) have limits so what SMBIOS are you using? Try one of the (other) Haswell plists.

**Comment by aa15032261** [comment]
I am using iMac16,2 (Mac-FFE5EF870D7BA81A) and I used freqVectorsEdit.sh to unlock 800MHz.

I also tried Mac Mini 6,2(Mac-F65AE981FFA204ED) and iMac15,1(Mac-42FD25EABCABB274) but still locked to 3.6GHz

**Comment by Piker-Alpha** [comment]
Strange. Perhaps the SSDT isn't loaded. 

If you don't know how to check this yourself then please e-mailing (see script for e-mail address) me your IORegistryExplorer (v2.0 only / use Google search) dump so that I can check some things for you.

**Comment by aa15032261** [comment]
I have sent you the file, thank you.

**Comment by Piker-Alpha** [comment]
Right. Your processor is locked (not an unlocked K variant) to 3.7GHz as maximum Turbo Frequency and that is what Apple uses – by reading certain MSRs – for power management.

You can probably also find errors in the macOS Sierra log output. Do this after a (re)boot.

```
sudo log config --mode "level:debug"
log show --predicate 'process == "kernel"' --debug --info --last "15m"
```

I also expect the output of AppleIntelInfo.kext to show some evidence of it. You may want to try it (download and compile it yourself).

**Comment by aa15032261** [comment]
I have sent you the debug output and AppleIntelInfo output to you email.

I found that Current Performance State Value is set to 3700MHz (which is my stock freq) and
Target performance State Value is 4000. However, I can run at 3600 only.

My cpu is not ending with 'K' though, C is just same as K (with an unlocked multiplier). Is there any way to bypass the lock?

**Comment by Piker-Alpha** [comment]
Thanks for the dump, but I forgot to mention that you should boot with: `debug=0x12a ioppf=0xfff` otherwise there is no XCPM related info.

**Comment by aa15032261** [comment]
Alright, I just sent you the output with the boot flag to your email.

**Comment by Piker-Alpha** [comment]
Got it. Thanks. The debug output clearly shows the problem:

kernel: (X86PlatformPlugin) X86PlatformPlugin::get_pstates_xcpm - 30 pstates, 4 turbo, 26 non-turbo
kernel: (X86PlatformPlugin) X86PlatformPlugin::get_pstates_xcpm - P 0: frequency 3700
../..
kernel: (X86PlatformPlugin) X86PlatformPlugin::get_pstates_xcpm - P29: frequency 800

And looking at the ACPI table output I see two possible problems:

kernel: (AppleACPIPlatform) SSDT 0x00000000D94F9000 000539 (v01 PmRef  Cpu0Ist  00003000 INTL 20051117)
kernel: (AppleACPIPlatform) SSDT 0x00000000D94F7000 000B74 (v01 CpuRef CpuSsdt  00003000 INTL 20051117)

Then a whole lot later the generated SSDT:

kernel: (AppleACPIPlatform) SSDT 0x00000000D94ED000 0008C0 (v01 APPLE  CpuPm    00020400 INTL 20140926)

In other words. If one of the previous tables include a P-State table, then the latter one won't be used, and that appears to be the case.

Now. I don't know if they are loaded dynamically or not, but you could try to drop them or look where they get loaded. Fact is. Your setup is failing due to ACPI configuration error.

You may zip the complete ACPI folder and attach it here so that I can have a look at it.

**Comment by mtgrosser** [comment]
Hi! Has there been any resolution to this issue? I'm having the same problem with i7-5775C on iMac 16,2, with max ratio = 40 enabled in BIOS. While the kernel log says

```kernel: (X86PlatformPlugin) X86PlatformPlugin::get_pstates_xcpm - 33 pstates, 7 turbo, 26 non-turbo
kernel: (X86PlatformPlugin) X86PlatformPlugin::get_pstates_xcpm - P 0: frequency 4000
kernel: (X86PlatformPlugin) X86PlatformPlugin::get_pstates_xcpm - P 1: frequency 3900
...
```

the processor still never exceeds 3.6 GHz:

```CPU Ratio Info:
------------------------------------
CPU Low Frequency Mode.............: 800 MHz
CPU Maximum non-Turbo Frequency....: 3300 MHz
CPU Maximum Turbo Frequency........: 4000 MHz
...
CPU P-States [ 8 10 12 14 15 17 18 19 20 21 22 23 24 26 27 28 29 30 31 32 33 34 35 36 (37) ] iGPU P-States [ (15) 16 19 20 21 23 ]
```
I tried Clover/DropTables/SSDT/Cpu0Ist and CpuSsdt, to no avail, so any help would be greatly appreciated! 

I attached the ACPI as well as the kernel log and AppleIntelInfo: [report.zip](https://github.com/Piker-Alpha/ssdtPRGen.sh/files/856751/report.zip)

**Comment by Piker-Alpha** [comment]
@mtgrosser,

I need the output of AppleIntelInfo.kext v2.5 not v1.2

**Comment by mtgrosser** [comment]
Thanks for looking into this, here is the output: [AppleIntelInfo-2.5.txt](https://github.com/Piker-Alpha/ssdtPRGen.sh/files/859358/AppleIntelInfo-2.5.txt)

**Comment by mtgrosser** [comment]
I did some more debugging, while the maximum P-state ever reached is still 37.

* When the method `_PR_.CPU0._DSM` is present, it is called, and XCPM enters registered mode. However, the `_PR_.CPU0.ACST`, `_PR_.CPU1.APSS` etc. methods from the generated SSDT are never called, no matter if the OEM PM SSDTs (`Cpu0Ist`, `CpuSsdt`, `ApIst`, `Cpu0Cst`, `ApCst`) are dropped or not.

* When XCPM is active, the plimit ratios are always identical to the max ratios set in BIOS. If max ratio is 40 in BIOS, 33 P-states are present in IOReg, with state 0 @ 4000 MHz and state 32 @ 800MHz.

* The same behaviour can be observed when replacing the generated SSDT with a minimal version which provides nothing but the `_PR_.CPU0._DSM` method in order to enable XCPM.

* When I remove the `_PR_.CPU0._DSM` method from the generated SSDT, the `_PR_.CPU0.ACST`, `_PR_.CPU1.APSS` etc. methods are called (but consequently, XCPM is off)

* Current Performance State Value is 0x2400 (3600 MHz), Target performance State Value is 0x2800 (4000 MHz), BIOS is unlocked

* Changing the system definition or importing other frequency vectors into the selected definition doesn't change anything.
